### PR TITLE
extensions: improve error message simplifying configuration object

### LINKED ISSUE #408: Improve Oauth2 configuration missing error message
At the moment, if I try to start a connector with the module `oauth2-core` without setting the property `edc.oauth.token.url` I get this error:
```
SEVERE 2021-12-13T13:39:02.286617 Error booting runtime
java.lang.NullPointerException: Missing Oauth2 config: TOKEN_URL
```

Instead, this should give a message like:
```
SEVERE 2021-12-13T13:39:02.286617 Error booting runtime
java.lang.NullPointerException: Missing Oauth2 config: edc.oauth.token.url
```

### PR DESCRIPTION
**Pull request by ndr-brt**
Fix #408

### CODE REVIEW

File: extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java [hunk 1]
```diff
@@ -173,6 +172,11 @@ private Oauth2Configuration createConfig(ServiceExtensionContext context) {
                 .privateKeyAlias(privateKeyAlias)
                 .clientId(clientId)
                 .privateKeyResolver(privateKeyResolver)
-                .certificateResolver(certificateResolver).build();
+                .certificateResolver(certificateResolver)
+                .build();
+    }
+
+    private String mandatorySetting(ServiceExtensionContext context, String key) {
+        return requireNonNull(context.getSetting(key, null), format("Missing %s mandatory config: %s", name(), key));
```
Comment (bscholtes1A) [inline extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java:180]: Seems weird to have the extension name in the middle of the error message, no?

**Review by bscholtes1A — APPROVED** [review]

File: extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java (hunk 1 above)
Comment (ndr-brt) [reply extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java:180]: Could be good to know which extension requires that parameter, e.g. to have the possibility to check documentation

Comment (ndr-brt) [reply extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java]: that's fair! done

**Review by bscholtes1A — APPROVED** [review]

**Review by MoritzKeppler — APPROVED** [review]
lgtm

**Pull request merged**